### PR TITLE
54 refactor nav bar

### PR DESCRIPTION
**Pull request by Cxristian**
update the documentation

### CODE REVIEW

File: src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutConfig.java [hunk 1]
```diff
@@ -41,7 +41,9 @@ public static PageLayoutConfig getInstance() {
      *
      * @return page layout.
      */
-    public String getPageLayoutName() {
-        return ReaderApplicationProperties.getInstance().getAppProperties().get(LAYOUT);
+    public static PageLayoutType getPageLayoutName() {
+        String layoutName = ReaderApplicationProperties.getInstance().getAppProperties().get(LAYOUT).toUpperCase();
+        return PageLayoutType.valueOf(layoutName);
     }
+
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutConfig.java]: delete this blank line please

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/NavBar.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * @version 0.0.1
+ */
+public abstract class NavBar extends BasePage {
+
+//    /**
+//     * Returns Account page after clicking on account option.
+//     *
+//     * @return an Account page.
+//     */
+//    public abstract AccountsPage clickAccountsOption();
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/NavBar.java]: delete or implement this method please

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/classic/common/NavBarClassic.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        wait.until(ExpectedConditions.visibilityOf(tabBar));
+    }
+
+//    /**
+//     * Returns the account classic page after clicking on account option.
+//     *
+//     * @return a account classic page.
+//     */
+//    @Override
+//    public AccountsClassicPage clickAccountsOption() {
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/classic/common/NavBarClassic.java]: delete this method or clear comment "//"

File: src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutConfig.java [hunk 4]
```diff
@@ -41,7 +41,9 @@ public static PageLayoutConfig getInstance() {
      *
      * @return page layout.
      */
-    public String getPageLayoutName() {
-        return ReaderApplicationProperties.getInstance().getAppProperties().get(LAYOUT);
+    public static PageLayoutType getPageLayoutName() {
+        String layoutName = ReaderApplicationProperties.getInstance().getAppProperties().get(LAYOUT).toUpperCase();
```
Comment (AKatrin) [inline src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutConfig.java]: It should be final

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/BaseAppLightningPage.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+     * Waits until page object is loaded.
+     */
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
+        wait.until(ExpectedConditions.visibilityOf(basePageHeader));
+    }
+
+    /**
+     * Gets for navBar.
+     * @return new navBar lighting.
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/BaseAppLightningPage.java]: Add a blank line above this line please

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/NavBarLightning.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        wait.until(ExpectedConditions.visibilityOf(tabBar));
+    }
+
+//    /**
+//     * Returns the account lightning page after clicking on account option.
+//     *
+//     * @return a account lightning page.
+//     */
+//    @Override
+//    public AccountsLightningPage clickAccountsOption() {
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/NavBarLightning.java]: delete or not comment this method please

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/NavBarLightning.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+//        accountsOption.click();
+//        return new AccountsLightningPage();
+//    }
+
+    /**
+     * Contact menu button.
+     */
+    public void clickContactOption() {
```
Comment (raultolive) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/lightning/common/NavBarLightning.java]: Change the name of method should be able by an Verb ejm: "openContactOption"

File: src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutFactory.java [hunk 8]
```diff
@@ -24,25 +30,74 @@
...
+     *
+     * @return a contact page.
+     */
+    public static ContactPageAbstract getNavigateContact() {
+        final ContactPageAbstract contactPageAbstract;
+        switch (PAGE_LAYOUT_TYPE) {
+            case CLASSIC:
+                contactPageAbstract = new ContactClassicPageAbstract();
+                break;
+            case LIGHTNING:
```
Comment (AKatrin) [inline src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutFactory.java]: Rename the method a getContactPage

File: src/main/java/org/fjala/gugumber/salesforce/ui/PageTransporter.java [hunk 9]
```diff
@@ -39,14 +40,14 @@
     private WebDriver webDriver;
 
     /**
-     * Variable for the name of page layout.
+     * Constant for the page layout type.
      */
-    private String pageLayoutName;
+    private PageLayoutType PAGE_LAYOUT_TYPE = PageLayoutConfig.getPageLayoutName();
```
Comment (AKatrin) [inline src/main/java/org/fjala/gugumber/salesforce/ui/PageTransporter.java]: It should be final and static

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/BaseAppPage.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
+package org.fjala.gugumber.salesforce.ui.pages.abstracts.common;
+
+import org.fjala.gugumber.salesforce.ui.BasePage;
+
+/**
+ * BaseAppPage class.
+ *
+ * @author Cxrisstian
```
Comment (AKatrin) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/BaseAppPage.java]: Fix your name

File: src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/BaseAppPage.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+ * @version 0.0.1
+ */
+public abstract class BaseAppPage extends BasePage {
+
+      public abstract NavBar getNavBar();
```
Comment (AKatrin) [inline src/main/java/org/fjala/gugumber/salesforce/ui/pages/abstracts/common/BaseAppPage.java]: Add documentation

File: src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutFactory.java (hunk 8 above)
Comment (Cxristian) [reply src/main/java/org/fjala/gugumber/salesforce/ui/PageLayoutFactory.java]: i don't understand where is the method called getContactPage

**Pull request merged**